cdptools/databases/document_store_database.py

```diff
@@ -0,0 +1,840 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Any, Dict, List, NamedTuple, Optional, Tuple, Union
+
+import pandas as pd
+
+from . import exceptions
+
+###############################################################################
+
+
+cdp_tables = [
+    "event",
+    "person",
+    "body",
+    "file",
+    "trasncript",
+    "seat",
+    "role",
+    "minutes_item",
+    "event_minutes_item",
+    "vote",
+    "matter",
+    "matter_type",
+]
+
+
+class WhereCondition(NamedTuple):
+    column_name: str
+    operator: str
+    value: Union[bool, datetime, float, int, str]
+
+
+class OrderCondition(NamedTuple):
+    column_name: str
+    operator: str
+
+
+class WhereOperators:
+    eq: str = "=="
+    contains: str = "in"
+    gt: str = ">"
+    lt: str = "<"
+    gteq: str = ">="
+    lteq: str = "<="
+
+
+class OrderOperators:
+    desc: str = "DESCENDING"
+    asce: str = "ASCENDING"
+
+
+class TermResult(NamedTuple):
+    term: str
+    contribution: float
+
+
+class Match:
+    def __init__(self, unique_id, terms: List[TermResult], data: Dict[str, Any]):
+        self._unique_id = unique_id
+        self._terms = terms
+        self._data = data
+
+    @property
+    def unique_id(self):
+        return self._unique_id
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def terms(self):
+        return self._terms
+
+    @property
+    def relevance(self):
+        return sum(t.contribution for t in self.terms)
+
+    def __str__(self):
+        return f"<Match [unique_id: {self.unique_id}, relevance: {self.relevance}]>"
+
+    def __repr__(self):
+        return str(self)
```

Comment:
Making a mental note for myself that we should move these out to it's own file probably.

Reply:
Yeah I realized this was redundant, I can make a separate PR moving this out for both the database classes